src/lib.rs

```diff
@@ -111,16 +158,27 @@ fn get_v4_header_and_payload(mut b: Bytes) -> Result<(Bytes, EncryptedPayload),
     }
 }
 
-pub fn decode_edoc(b: Bytes) -> Result<(V4DocumentHeader, EncryptedPayload), Error> {
-    let (header_bytes, attached_document) = get_v4_header_and_payload(b)?;
-
-    let pb = protobuf::Message::parse_from_bytes(&header_bytes[..])
-        .map_err(|e| Error::HeaderParseErr(e.to_string()))?;
-    Ok((pb, attached_document))
+pub fn create_signed_header(
+    edek_wrapper: icl_header_v4::v4document_header::EdekWrapper,
+    signing_key: EncryptionKey,
+) -> V4DocumentHeader {
+    let signed_payload = icl_header_v4::v4document_header::SignedPayload {
+        edeks: vec![edek_wrapper],
+        ..Default::default()
+    };
+    let signature_info = sign_header(signing_key.0, &signed_payload);
+    icl_header_v4::V4DocumentHeader {
+        signed_payload: Some(signed_payload).into(),
+        signature_info: Some(signature_info).into(),
+        ..Default::default()
+    }
 }
 
-/// Construct an IronCore EDOC from the constituent parts.
-pub fn encode_edoc(header: V4DocumentHeader, payload: EncryptedPayload) -> Result<Bytes, Error> {
+/// Construct an IronCore attached EDOC from the constituent parts.
+pub fn encode_edoc(
```

Comment:
This and the decode should be something like `encode_attached_edoc` to help inform future callers so they don't try to mix use cases.

Reply:
The v4 EDOC header in the detached case still exists yes? It's wrapping the EDEK and is detached, but it exists. I don't see the reason to _not_ include attached in the name here.

Reply:
I addded the attached word to ensure it's clear.